### PR TITLE
Fix the inconsistency of composed properties

### PR DESCRIPTION
**Pull request by mdiep**
This is an alternative to #3042 that I think is quite a bit simpler. cc @andersio 

The problem, restated:

```swift
let A = MutableProperty(1)
let B = MutableProperty(100)
let AB = A.zip(with: B)
A.value = 2

// This mistakenly prints 102 because even though `AB.value` is 101,
// the producer is passed through. So when `AB.producer` is used,
// the initial producer is started again, yielding a potentially different
// set of zipped values.
AB.producer.startWithNext { print($0) }
```

This PR uses a replayed producer to provide a consistent value.

### CODE REVIEW

File: ReactiveCocoa/Swift/Property.swift [hunk 1]
```diff
@@ -481,88 +481,38 @@ public final class Property<Value>: PropertyProtocol {
...
-			var signal: Signal<Value, NoError>!
-			atomicSignal.modify { innerSignal in
-				if signal == nil {
-					innerSignal = prepareRelaySignal(relay.signal)
-				}
-				signal = innerSignal
-			}
-			return signal
+			var extractedSignal: Signal<Value, NoError>!
+			producer.skip(first: 1).startWithSignal { signal, _ in extractedSignal = signal }
```
Comment (andersio) [inline ReactiveCocoa/Swift/Property.swift]: `skip(first: 1)` is not necessary. The current value would be sent right after the `setUp` closure is invoked.

Comment (mdiep) [inline ReactiveCocoa/Swift/Property.swift]: Ah, right.

File: ReactiveCocoa/Swift/Property.swift [hunk 2]
```diff
@@ -481,88 +481,38 @@ public final class Property<Value>: PropertyProtocol {
...
-
-		_value = { relay.value! }
-		_producer = { prepareRelayProducer(relay.producer) }
-
-		// Lazily initializes the signal of this composed property.
-		//
-		// The created signal would retain `scopedDisposable` to keep the relay
-		// alive, thus inevitably binding the relay to the lifetime of the upstream
-		// for good.
-		let atomicSignal = Atomic<Signal<Value, NoError>?>(nil)
```
Comment (andersio) [inline ReactiveCocoa/Swift/Property.swift]: This is still a problem unless we change the semantics of `Signal`. Uniquing the signal at least limited the scope of *potential* leaking.

Comment (andersio) [inline ReactiveCocoa/Swift/Property.swift]: Consider this snippet:
```
let property = source.map { $0 + 1 }
for i in 0 ..< 100000 { _ = property.signal }
```

100000 signals would be inevitably kept alive until `source` deinitializes...

Comment (mdiep) [inline ReactiveCocoa/Swift/Property.swift]: Want to open an issue to consider that separately? I think that ties into our discussion about the lifetime semantics of `Signal`.

**Comment by andersio** [comment]
LGTM with a nitpick on the change to `signal`. It is pretty awesome to see `replayLazily` fit in. Haven't thought of it before. :tada:

File: ReactiveCocoa/Swift/Property.swift (hunk 2 above)
Comment (andersio) [inline ReactiveCocoa/Swift/Property.swift]: Sure. But I just think we should stick with the status quo of the branch, shouldn't we? We might end up not changing the semantics.

Comment (mdiep) [inline ReactiveCocoa/Swift/Property.swift]: That's why I'd open a separate issue: if we don't change the semantics, we'll have an issue that says we should fix this.

Comment (andersio) [inline ReactiveCocoa/Swift/Property.swift]: Okay.

**Comment by mdiep** [comment]
I think this is ready to be merged then?

**Pull request merged**